db/music_pg.sql

```diff
@@ -0,0 +1,373 @@
+CREATE TABLE "submission" (
+  "submission_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "submission_history" (
+  "submission_history_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "date_update" timestamp NOT NULL,
+  "note_update" text NOT NULL,
+  "submission_id" int NOT NULL,
+  "participant_id" int NOT NULL,
+  "bracket_id" int NOT NULL,
+  "song_id" int NOT NULL,
+  "order" int NOT NULL,
+  "date_submitted" timestamp NOT NULL DEFAULT (now()),
+  "active" bool_int NOT NULL,
+  "is_dupe" bool_int NOT NULL,
+  "title" text NOT NULL,
+  "version" text,
+  "artist" text,
+  "collaborators" text,
+  "composer" text,
+  "link" text NOT NULL,
+  "genre" text,
+  "emoji" text,
+  "comments" text
+);
+
+CREATE TABLE "link_relation" (
+  "link_realtion_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "relation_id" int NOT NULL,
+  "link_id" int NOT NULL,
+  "type_link" text NOT NULL
+);
+
+CREATE TABLE "link" (
+  "link_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "url" text NOT NULL,
+  "region" text[]
+);
+
+CREATE TABLE "song" (
+  "song_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "title" text NOT NULL,
+  "title_english" text,
+  "version" text,
+  "artist" text NOT NULL,
+  "collaborators" text,
+  "composer" text,
+  "year_min" int,
+  "year_max" int
+);
+
+CREATE TABLE "song_relation" (
+  "song_relation_id" INT GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
+  "song_from_id" int NOT NULL,
+  "song_to_id" int NOT NULL,
+  "relation_type" text NOT NULL
+);
+
+CREATE TABLE "song_artist" (
+  "song_artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "song_id" int NOT NULL,
+  "artist_id" int NOT NULL,
+  "credit_type" text NOT NULL
+);
+
+CREATE TABLE "artist" (
+  "artist_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "name" text NOT NULL,
+  "year_min" int,
+  "year_max" int,
+  "primary_artist_id" int
+);
+
+CREATE TABLE "artist_genre" (
+  "artist_genre_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "artist_id" int NOT NULL,
+  "genre_id" int NOT NULL
+);
+
+CREATE TABLE "participant" (
+  "participant_id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
+  "username" text UNIQUE NOT NULL,
+  "nickname" text,
+  "date_joined" timestamp NOT NULL DEFAULT (now()),
+  "time_zone" text,
+  "active" bool_int NOT NULL
+);
+
+CREATE TABLE "participant_nickname" (
```

Comment:
If we have this table, should we *also* have the `nickname` column on the `participant` table or should we just join to this table and grab the most recent one?

Reply:
The latter. Trying not to duplicate data/give us more places to update. Unless we make it a calculated column.